dist/tools/whitespacecheck/check.sh

```diff
@@ -27,8 +27,14 @@ if [ -z "${BRANCH}" ]; then
     BRANCH=$(git rev-list HEAD | tail -n 1)
 fi
 
-git diff --check $(git merge-base ${BRANCH} HEAD)
-if [ $? -gt 0 ]
+git -c core.whitespace="tab-in-indent,tabwidth=4" \
+    diff --check $(git merge-base ${BRANCH} HEAD) -- *.[ch]
+
+RESULT=$?
+
+git -c core.whitespace="trailing-space" \
+    diff --check $(git merge-base ${BRANCH} HEAD)
```

Comment:
(I would be confused by the by the preceiding `-` in `-cr-at-eol` in a few month from now I guess, because I know that git uses UNIX line endings, would read it as "accept cr line endings", then would go into the doc, stumble upon the `-` bit and would have wasted 5 minutes of my life :P)